src/Raven.Server/RavenServer.cs

```diff
@@ -931,39 +937,105 @@ internal AuthenticateConnection AuthenticateConnectionCertificate(X509Certificat
+                        var cert = ServerStore.Cluster.Read(ctx, certKey) ??
+                                   ServerStore.Cluster.GetLocalState(ctx, certKey);
+
+                        if (cert == null)
                         {
-                            authenticationStatus.Status = AuthenticationStatus.Operator;
+                            // The certificate is not explicitly registered in our server, let's see if we have a certificate
+                            // with the same public key pinning hash.
+                            var allCertKeys = ServerStore.Cluster.ItemKeysStartingWith(ctx, Constants.Certificates.Prefix, 0, int.MaxValue).ToList();
+                            allCertKeys.AddRange(ServerStore.Cluster.GetCertificateKeysFromLocalState(ctx));
+
+                            foreach (var key in allCertKeys)
+                            {
+                                var currentCert = ServerStore.Cluster.Read(ctx, key) ??
+                                                  ServerStore.Cluster.GetLocalState(ctx, key);
+
+                                if (currentCert == null)
+                                    continue;
+
+                                if (currentCert.TryGet(nameof(CertificateDefinition.PublicKeyPinningHash), out string hash) == false)
+                                    continue;
+
+                                if (CertificateUtils.GetPublicKeyPinningHash(certificate).Equals(hash) == false)
+                                    continue;
+
+                                if (currentCert.TryGet(nameof(CertificateDefinition.Thumbprint), out string thumbprint) && certificate.Thumbprint.Equals(thumbprint))
+                                    continue;
+                                // Different certificate but same public key pinning hash. We'll allow the connection and register the used certificate
+                                // in the server with the same permissions as the original.
+                                var currentCertDef = JsonDeserializationServer.CertificateDefinition(currentCert);
+                                var certBytes = certificate.Export(X509ContentType.Cert);
+
+                                var newCopy = new CertificateDefinition
+                                {
+                                    Name = currentCertDef.Name,
+                                    Certificate = Convert.ToBase64String(certBytes),
+                                    Permissions = currentCertDef.Permissions,
+                                    SecurityClearance = currentCertDef.SecurityClearance,
+                                    Thumbprint = certificate.Thumbprint,
+                                    PublicKeyPinningHash = CertificateUtils.GetPublicKeyPinningHash(certificate),
+                                    NotAfter = certificate.NotAfter,
+                                };
+
+                                lock (_certificatesLock)
```

Comment:
Why? This isn't the right place to handle the actual logic of this.
You detected, based on your current state, that the certificate should be added.
Now you call to `PutCertificateCollectionInCluster`, which is going to add it to the cluster, which already make it a sync operation.
Use a dedicated command for this, that will handle  the check if it exists, remove previous copies, etc.
Note that you should also be explicit that the certificate has already been replaced, so registering this in the cluster may _fail_, and should be handled.

Reply:
Added a dedicated command.